Ladybird/HelperProcess.cpp

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2023, Andrew Kaster <akaster@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "HelperProcess.h"
+#include "Utilities.h"
+#include <AK/String.h>
+#include <QCoreApplication>
+
+ErrorOr<void> spawn_helper_process(StringView process_name, Span<StringView> arguments, Core::System::SearchInPath search_in_path, Optional<Span<StringView>> environment)
+{
+    auto paths = TRY(get_paths_for_helper_process(process_name));
+    VERIFY(!paths.is_empty());
+    ErrorOr<void> result;
+    for (auto const& path : paths) {
+        arguments[0] = path.bytes_as_string_view();
+        result = Core::System::exec(path, arguments, search_in_path, environment);
+        if (!result.is_error())
+            break;
+    }
+
+    return result;
+}
+
+ErrorOr<Vector<String>> get_paths_for_helper_process(StringView process_name)
+{
+    Vector<String, 2> paths;
```

Comment:
Seems odd to have an inline storage of 2, and then add 3 paths? Probably fine to just not have inline storage at all here

Reply:
Good point. I previously had 2 paths here and it made a bit more sense. removed the inline capacity.